internal/app/wwctl/overlay/delete/main.go

```diff
@@ -10,88 +10,94 @@ import (
 	"github.com/hpcng/warewulf/internal/pkg/overlay"
 	"github.com/hpcng/warewulf/internal/pkg/util"
 	"github.com/hpcng/warewulf/internal/pkg/wwlog"
+	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 )
 
 func CobraRunE(cmd *cobra.Command, args []string) error {
 	var overlayPath string
+	var fileName string
 
-	if SystemOverlay {
-		overlayPath = config.SystemOverlaySource(args[0])
-	} else {
-		overlayPath = config.RuntimeOverlaySource(args[0])
+	overlayKind := args[0]
+	overlayName := args[1]
+
+	if len(args) == 3 {
+		fileName = args[2]
+	}
+
+	if overlayKind != "system" && overlayKind != "runtime" {
+		return errors.New("overlay kind must be of type 'system' or 'runtime'")
+	}
+
+	if overlayKind == "system" {
+		overlayPath = config.SystemOverlaySource(overlayName)
+	} else if overlayKind == "runtime" {
+		overlayPath = config.RuntimeOverlaySource(overlayName)
 	}
 
 	if overlayPath == "" {
-		wwlog.Printf(wwlog.ERROR, "Overlay name did not render: '%s'\n", args[0])
+		wwlog.Printf(wwlog.ERROR, "Overlay name did not resolve: '%s'\n", overlayName)
 		os.Exit(1)
 	}
 
 	if !util.IsDir(overlayPath) {
-		wwlog.Printf(wwlog.ERROR, "Overlay name does not exist: '%s'\n", args[0])
+		wwlog.Printf(wwlog.ERROR, "Overlay does not exist: '%s:%s'\n", overlayKind, overlayName)
 		os.Exit(1)
 	}
 
-	if len(args) == 1 {
+	if fileName == "" {
 		if Force {
 			err := os.RemoveAll(overlayPath)
 			if err != nil {
-				wwlog.Printf(wwlog.ERROR, "Failed deleting overlay: %s\n", args[0])
-				wwlog.Printf(wwlog.ERROR, "%s\n", err)
-				os.Exit(1)
+				return errors.Wrap(err, "failed deleting overlay")
 			}
 		} else {
 			err := os.Remove(overlayPath)
 			if err != nil {
-				wwlog.Printf(wwlog.ERROR, "Failed deleting overlay: %s\n", args[0])
-				wwlog.Printf(wwlog.ERROR, "%s\n", err)
-				os.Exit(1)
+				return errors.Wrap(err, "failed deleting overlay")
 			}
 		}
 		fmt.Printf("Deleted overlay: %s\n", args[0])
 
-	} else if len(args) > 1 {
-		for i := 1; i < len(args); i++ {
-			removePath := path.Join(overlayPath, args[i])
+	} else {
+		removePath := path.Join(overlayPath, overlayPath)
```

Comment:
I think we need `removePath := path.Join(overlayPath, fileName)` here.

I am getting this for a path:
```
[ERROR]           Path to remove doesn't exist in overlay: /var/warewulf/overlays/runtime/myOverlay1/var/warewulf/overlays/runtime/myOverlay1
```

Reply:
I concur. Fixing.